contracts/CometCore.sol

```diff
@@ -69,54 +69,53 @@ abstract contract CometCore is CometConfiguration, CometStorage, CometMath {
 
     /**
      * @dev The positive present supply balance if positive or the negative borrow balance if negative
-     *  Note This will overflow at 2^103/1e18=~10 trillion for assets with 18 decimals.
      */
-    function presentValue(int104 principalValue_) internal view returns (int104) {
+    function presentValue(int104 principalValue_) internal view returns (int256) {
         if (principalValue_ >= 0) {
-            return signed104(presentValueSupply(baseSupplyIndex, unsigned104(principalValue_)));
+            return signed256(presentValueSupply(baseSupplyIndex, uint104(principalValue_)));
         } else {
-            return -signed104(presentValueBorrow(baseBorrowIndex, unsigned104(-principalValue_)));
+            return -signed256(presentValueBorrow(baseBorrowIndex, uint104(-principalValue_)));
         }
     }
 
     /**
      * @dev The principal amount projected forward by the supply index
-     *  Note This will overflow at 2^104/1e18=~20 trillion for assets with 18 decimals.
      */
-    function presentValueSupply(uint64 baseSupplyIndex_, uint104 principalValue_) internal pure returns (uint104) {
-        return uint104(uint(principalValue_) * baseSupplyIndex_ / BASE_INDEX_SCALE);
+    function presentValueSupply(uint64 baseSupplyIndex_, uint104 principalValue_) internal pure returns (uint256) {
+        return uint256(principalValue_) * baseSupplyIndex_ / BASE_INDEX_SCALE;
     }
 
     /**
      * @dev The principal amount projected forward by the borrow index
-     *  Note This will overflow at 2^104/1e18=~20 trillion for assets with 18 decimals.
      */
-    function presentValueBorrow(uint64 baseBorrowIndex_, uint104 principalValue_) internal pure returns (uint104) {
-        return uint104(uint(principalValue_) * baseBorrowIndex_ / BASE_INDEX_SCALE);
+    function presentValueBorrow(uint64 baseBorrowIndex_, uint104 principalValue_) internal pure returns (uint256) {
+        return uint256(principalValue_) * baseBorrowIndex_ / BASE_INDEX_SCALE;
     }
 
     /**
      * @dev The positive principal if positive or the negative principal if negative
      */
```

Comment:
nit: should we add the same `Note: This will overflow...` here as well?

Reply:
I actually don't think so personally, easier to maintain if we keep it close to the issue IMO